packages/langium-cli/src/generator/ast-generator.ts

```diff
@@ -234,3 +236,22 @@ function groupBySupertypes(astTypes: AstTypes): MultiMap<string, string> {
 
     return superToChild;
 }
+
+function generateTerminalConstants(fileNode: CompositeGeneratorNode, grammars: Grammar[]) {
+    let collection: Record<string, RegExp> = {};
+    grammars.forEach(grammar => {
+        const terminalConstants = collectTerminalRegexps(grammar);
+        collection = {...collection, ...terminalConstants};
+    });
+
+    fileNode.append(`export type TerminalNames = ${Object.keys(collection).map(name => `'${name}'`).join(' | ')};`, NL, NL);
+
+    fileNode.append('export const TerminalRegExps: Record<TerminalNames, RegExp> = {', NL);
```

Comment:
We should actually only generate the following:

```ts
export const LangiumGrammarTerminals = {
    ID: /\^?[_a-zA-Z][\w_]*/,
    STRING: /"(\\.|[^"\\])*"|'(\\.|[^'\\])*'/,
    RegexLiteral: /\/(?![*+?])(?:[^\r\n\[/\\]|\\.|\[(?:[^\r\n\]\\]|\\.)*\])+\/[a-z]*/,
    WS: /\s+/,
    ML_COMMENT: /\/\*[\s\S]*?\*\//,
    SL_COMMENT: /\/\/[^\n\r]*/,
};
```

If adopters want to have the corresponding `TerminalNames` type, they can just use `keyof typeof LangiumGrammarTerminals`. Note that this also encodes the project name, just to prevent any conflicts between multiple Langium projects being imported in a file.